src/components/organisms/friend-list/ListFriend.tsx

```diff
@@ -3,23 +3,45 @@ import { useRouter } from 'next/router';
 import * as S from './styled';
 import FRIENDS, { Friend } from '@/apis/friend-api/friendList';
 import { useRecoilState } from 'recoil';
-import { friendInfoState } from '@/recoil/atoms/FriendsAtom';
+import { FriendInfo, friendInfoState } from '@/recoil/atoms/FriendsAtom';
 
 const ListFriend = () => {
   const router = useRouter();
   const [friend, setFriend] = useState<Friend['data']>([]);
-  const [isDelete, setIsDelete] = useState(false);
   const [friendInfo, setFriendInfo] = useRecoilState(friendInfoState);
+  const loginUserId: number | undefined = router.query.id
+    ? Number(router.query.id)
+    : undefined;
 
   const getListFriend = async () => {
     try {
       const response = await FRIENDS.getFriendList();
       setFriend(response);
-      const updatedFriendInfo = response.map((item: any) => ({
-        requesterId: item.requesterId || 0,
-        name: item.requester?.name || '',
-        image: item.requester?.userImage?.imageUrl || '',
-      }));
+      const updatedFriendInfo: FriendInfo[] = response.map((item: any) => {
+        const isCurrentUserRequester = loginUserId === item.requesterId;
+
+        return {
+          requesterId: isCurrentUserRequester
+            ? item.requesterId
+            : item.respondentId,
+          requesterName: isCurrentUserRequester
+            ? item.requester?.name || ''
+            : item.respondent?.name || '',
+          requesterImage: isCurrentUserRequester
+            ? item.requester?.userImage?.imageUrl || ''
+            : item.respondent?.userImage?.imageUrl || '',
+          respondentId: isCurrentUserRequester
+            ? item.respondentId
+            : item.requesterId,
+          respondentName: isCurrentUserRequester
+            ? item.respondent?.name || ''
+            : item.requester?.name || '',
+          respondentImage: isCurrentUserRequester
+            ? item.respondent?.userImage?.imageUrl || ''
+            : item.requester?.userImage?.imageUrl || '',
+        };
+      });
```

Comment:
비~~슷한 코드들이 반복되는 느낌이에요 이 부분 어떻게 개선하면 좋을까요? 🤔
아래 예제처럼 `getUserType` 함수를 분리하고, `객체 대괄호 접근법`으로 관리하면 조금 더 깔끔해지고 가독성을 높일 수 있을 것 같아요 😄
또한, map은 새로운 배열을 반환해주기 때문에 `updatedFriendInfoList` 라고 네이밍을 지어봤어요
간단한 아이디어이니 참고해보시고 타입은 제가 정확히 몰라서 작성하지 않았으니 감안해주세요
네이밍도 조금 더 적절한게 무엇일까 고민해보면 좋을 것 같네요 😄
```suggestion
const getUserType = (loginUserId, requesterId) => {
  const isCurrentUserRequester = loginUserId === requesterId;

  return {
    requester: isCurrentUserRequester ? "requester" : "respondent",
    respondent: isCurrentUserRequester ? "respondent" : "requester",
  };
};

const updatedFriendInfoList = response.map((item) => {
  const { requester, respondent } = getUserType(loginUserId, item.requesterId);

  return {
    requesterId: item[`${requester}Id`],
    requesterName: item[requester]?.name || "",
    requesterImage: item[requester]?.userImage?.imageUrl || "",
    respondentId: item[`${respondent}Id`],
    respondentName: item[respondent]?.name || "",
    respondentImage: item[respondent]?.userImage?.imageUrl || "",
  };
});
```